dashboard/app/controllers/vocabularies_controller.rb

```diff
@@ -36,11 +36,25 @@ def update
     end
   end
 
+  # GET /courses/:course_name/vocab/edit
+  def edit
+    @course_version = find_matching_course_version
+    @vocabularies = @course_version.vocabularies.order(:word)
+  end
+
   private
 
   def vocabulary_params
     vp = params.transform_keys(&:underscore)
     vp = vp.permit(:id, :key, :word, :definition, :course_version_id)
     vp
   end
+
+  def find_matching_course_version
+    matching_unit_group = UnitGroup.find_by_name(params[:course_name])
+    return matching_unit_group.course_version if matching_unit_group
+    matching_standalone_course = Script.find_by_name(params[:course_name])
+    return matching_standalone_course&.get_course_version if matching_standalone_course&.is_course
```

Comment:
on this line in particular, I think it might be better to do:
```suggestion
return matching_standalone_course&.course_version if matching_standalone_course&.is_course
```
I think this will do the same thing as the code you have here, but it's a bit more straight forward because we are already checking that we're handling the case where this script is not in a unit group.